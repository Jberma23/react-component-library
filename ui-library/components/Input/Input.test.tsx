import { render } from "@testing-library/react";
import { Input } from ".";

describe("ui-library/components/Button/Button.tsx", () => {
  it("renders with defaults and no label", () => {
    const { asFragment } = render(<Input />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders with defaults and a label", () => {
    const { asFragment } = render(<Input label={"My Label"} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders when disabled", () => {
    const { asFragment } = render(<Input label={"My Label"} disabled />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders with fullWidth", () => {
    const { asFragment } = render(<Input label={"My Label"} fullWidth />);
    expect(asFragment()).toMatchSnapshot();
  });
});
