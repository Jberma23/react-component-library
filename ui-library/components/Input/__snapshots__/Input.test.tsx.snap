// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-library/components/Button/Button.tsx renders when disabled 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-xxs min-w-0"
  >
    <label
      for=":r2:"
    >
      My Label
    </label>
    <input
      class="border-1 border-gray-300 rounded-3 p-xs focus-visible:outline focus-visible:outline-teal-500 focus-visible:border-teal-500 focus-visible:outline-offset-0 outline-offset-0 transition-all duration-fast bg-gray-100 cursor-not-allowed"
      disabled=""
      id=":r2:"
    />
  </div>
</DocumentFragment>
`;

exports[`ui-library/components/Button/Button.tsx renders with defaults and a label 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-xxs min-w-0"
  >
    <label
      for=":r1:"
    >
      My Label
    </label>
    <input
      class="border-1 border-gray-300 rounded-3 p-xs focus-visible:outline focus-visible:outline-teal-500 focus-visible:border-teal-500 focus-visible:outline-offset-0 outline-offset-0 transition-all duration-fast"
      id=":r1:"
    />
  </div>
</DocumentFragment>
`;

exports[`ui-library/components/Button/Button.tsx renders with defaults and no label 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-xxs min-w-0"
  >
    <input
      class="border-1 border-gray-300 rounded-3 p-xs focus-visible:outline focus-visible:outline-teal-500 focus-visible:border-teal-500 focus-visible:outline-offset-0 outline-offset-0 transition-all duration-fast"
      id=":r0:"
    />
  </div>
</DocumentFragment>
`;

exports[`ui-library/components/Button/Button.tsx renders with fullWidth 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-xxs min-w-0 w-full"
  >
    <label
      for=":r3:"
    >
      My Label
    </label>
    <input
      class="border-1 border-gray-300 rounded-3 p-xs focus-visible:outline focus-visible:outline-teal-500 focus-visible:border-teal-500 focus-visible:outline-offset-0 outline-offset-0 transition-all duration-fast w-full"
      id=":r3:"
    />
  </div>
</DocumentFragment>
`;
